File: client/src/types/salaryHistory.ts
export interface SalaryHistory {
  id: string;
  employeeId: string;
  amount: number;
  effectiveFrom: string;
  reason?: string | null;
  createdAt: string;
}

export interface AddSalaryHistoryDto {
  employeeId: string;
  amount: number;
  effectiveFrom: string;
  reason?: string;
}

File: EmployeeAdminPortal/EmployeeAdminPortal/Models/Entities/SalaryHistory.cs
namespace EmployeeAdminPortal.Models.Entities
{
    public class SalaryHistory
    {
        public Guid Id { get; set; }
        public Guid EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public decimal Amount { get; set; }
        public DateTime EffectiveFrom { get; set; }
        public string? Reason { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
